### ISSUE TITLE
Indentation check fails for switch statements

### ISSUE THREAD

**Issue by liutikas** [checkstyle/checkstyle#341]
Configuration file:

``` xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="severity" value="error"/>
      <property name="basicOffset" value="4"/>
      <property name="throwsIndent" value="8"/>
      <property name="lineWrappingIndentation" value="8"/>
    </module>
  </module>
</module>
```

Java Class:

``` java
public class MyClass {
    public int getValue(int value) {
        switch (value) {
            case 0: return ABC1;
            case 1: return ABC2;
            case 2: return ABC3;
        }
        return 0;
    }
}
```

Checkstyle warns with:

```
MyClass.java:4: 'block' child have incorrect indentation level 12, expected level should be 16.
MyClass.java:5: 'block' child have incorrect indentation level 12, expected level should be 16.
MyClass.java:6: 'block' child have incorrect indentation level 12, expected level should be 16.
```

If I update the Java lines 4, 5, and 6 to have 16 spaces instead and then I get:

```
MyClass.java:4: 'case' child have incorrect indentation level 16, expected level should be 12.
MyClass.java:5: 'case' child have incorrect indentation level 16, expected level should be 12.
MyClass.java:6: 'case' child have incorrect indentation level 16, expected level should be 12.
```

Thoughts?

Ideally some other module would let you know about need for a new line or something.
